keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensor.java

```diff
@@ -81,7 +87,7 @@ public static Nd4jIntegerTensor zeros(long[] shape) {
     }
 
     static INDArray unsafeGetNd4J(IntegerTensor that) {
-        if (that.isLengthOne()) {
+        if (that.isScalar() || that instanceof ScalarIntegerTensor) {
```

Comment:
if `that` is an instance of `ScalarIntegerTensor` shouldn't it return `true` on `that.isScalar()`?

Reply:
Nope, scalars are rank 0, while `Scalar...Tensor`s can have any rank (a shape of `[1,1,...,1]`). It is confusing though, so maybe we should think about renaming `isScalar`?